File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap();
        
        for(int i = 0; i < nums.length; i++ ){
            map.put(nums[i],i);
        }
        
        int[] ans = new int[2];
        
        for(int i = 0; i< nums.length; i++){
            int val = target-nums[i];
            if(map.containsKey(val) && map.get(val) != i){
                ans[0] = i;
                ans[1] = map.get(val);
                break;
            }
        }
        
        return ans;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> v;
        int f = 0;
        for(int i=0;i<nums.size();i++){
            for(int j=0;j<nums.size();j++){
                if(j==i)
                    continue;
                if((nums.at(i)+nums.at(j))==target && f == 0){
                    v.push_back(i);
                    v.push_back(j);
                    f = 1;
                }
            }
        }
        return v;
    }
};